src/controller/p2p/preheat/controller.go

```diff
@@ -222,11 +256,77 @@ func (c *controller) UpdatePolicy(ctx context.Context, schema *policyModels.Sche
 	if schema != nil {
 		schema.UpdatedTime = time.Now()
 	}
+	s0, err := c.pManager.Get(ctx, schema.ID)
+	if err != nil {
+		return err
+	}
+	var cron = schema.Trigger.Settings.Cron
+	var oldJobID = s0.Trigger.Settings.JobID
+	var needUn bool
+	var needSch bool
+
+	if s0.Trigger.Type != schema.Trigger.Type {
+		if s0.Trigger.Type == policyModels.TriggerTypeScheduled && oldJobID > 0 {
+			needUn = true
+		}
+		if schema.Trigger.Type == policyModels.TriggerTypeScheduled && len(cron) > 0 {
+			needSch = true
+		}
+	} else {
+		// not change trigger type
+		switch schema.Trigger.Type {
+		case policyModels.TriggerTypeScheduled:
+			if s0.Trigger.Settings.Cron != cron {
+				// unschedule old
+				if oldJobID > 0 {
+					needUn = true
+				}
+				// schedule new
+				if len(cron) > 0 {
+					// valid cron
+					needSch = true
+				}
+			}
+
+		case "":
+		default:
+			return fmt.Errorf("not support Trigger %s", schema.Trigger.Type)
+		}
+	}
+
+	// unschedule old
+	if needUn {
+		err = c.scheduler.UnSchedule(ctx, oldJobID)
+		if err != nil {
```

Comment:
How can we handle the old jobID

Reply:
if `UnSchedule` the old-one success but `Schedule` the new-one failed, user will always get error when retry update, because the old-one's `jobID` can not found. Can we return success if `jobID` not found when call `UnSchedule`?